### ISSUE TITLE
Error when generating bigwig format

### ISSUE THREAD

**Issue by baijiexu** [fl-yu/CUT-RUNTools-2.0#56]
Hi Fulong,

Can I ask for a help for the error during generating bigwig format?

Here it is,
`[info] Generating the normalized signal file with BigWig format...
Thu Jul 21 23:22:56 CDT 2022
cp: cannot stat ‘/endosome/work/InternalMedicine/s207732/_CUTRUN/_Sim1cre/_set1/pos/positive/peakcalling/macs2.narrow/positive.cpm.norm.bw’: No such file or directory
cp: cannot stat ‘/endosome/work/InternalMedicine/s207732/_CUTRUN/_Sim1cre/_set1/pos/positive/peakcalling/macs2.narrow/positive.cpm.norm.bw’: No such file or directory
`

`usage: An example usage is:$ bamCoverage -b reads.bam -o coverage.bw
bamCoverage: error: unrecognized arguments: CPM --effectiveGenomeSize 2652783500`

And I checked my config files, the path for deeptools is correct. I have no idea what's the problem now. Do you have any idea about it? Many many thanks and have a good day!

Baijie

**Comment by baijiexu** [comment]
Hi Fulong,

Again me. When I ran the script one by one, I found the error for bamCoverage.

`GATA1_D7_30min_chr11.bam does not appear to have an index. You MUST index the file first!`

Do you think this is the problem? However, I didn't find the script of index for this bam file. 

Best,
Baijie

**Comment by fl-yu** [comment]
Hi Baijie, 

Thank you for your interest in our tool!
It looks that the bam file was generated (you should check this) and the issue is perhaps about deeptools. I think deeptools could work while the parameter can not be recognized. 
Can you check the version of deeptools?
Can you run the following command? Please replace `$bam_file` with your real bam file, which should be found in the deduplicated folder.

```
bamCoverage --bam $bam_file -o /endosome/work/InternalMedicine/s207732/_CUTRUN/_Sim1cre/_set1/pos/positive/peakcalling/macs2.narrow/positive.cpm.norm.bw \
    --binSize 500 \
    --normalizeUsing CPM \
    --effectiveGenomeSize 2652783500
```

**Comment by baijiexu** [comment]
Hi Fulong,

Thank you so much for your reply. 

`deeptools 2.5.7` . This is my version of deeptools.

Actually, I tried this bamCoverage this afternoon using the real indexed bam file. It works. However, I couldn't understand why it doesn't work in the pipeline.

Baijie

**Comment by baijiexu** [comment]
Hi Fulong,

Again me. The thing is here.

```
bamCoverage --bam GATA1_D7_30min_chr11.bam -o /home2/s207732/CUT_RUNTools-2.0/bulk-example-test/GATA1_D7_30min_chr11/peakcalling/macs2.narrow/GATA1_D7_30min_chr11.cpm.norm.bw
bamFilesList: ['GATA1_D7_30min_chr11.bam']
binLength: 50
numberOfSamples: None
blackListFileName: None
defaultFragmentLength: read length
numberOfProcessors: 16
verbose: False
region: None
bedFile: None
minMappingQuality: None
ignoreDuplicates: False
chrsToSkip: []
stepSize: 50
center_read: False
samFlag_include: None
samFlag_exclude: None
minFragmentLength: 0
maxFragmentLength: 0
zerosToNans: False
smoothLength: None
save_data: False
out_file_for_raw_data: None
maxPairedFragmentLength: 1000
```
When I add the parameters,
```
bamCoverage --bam GATA1_D7_30min_chr11.bam -o /home2/s207732/CUT_RUNTools-2.0/bulk-example-test/GATA1_D7_30min_chr11/peakcalling/macs2.narrow/GATA1_D7_30min_chr11.cpm.norm.bw --binSize 10 --normalizeUsing CPM --effectiveGenomeSize 2652783500
usage: An example usage is:$ bamCoverage -b reads.bam -o coverage.bw
bamCoverage: error: unrecognized arguments: CPM --effectiveGenomeSize 2652783500
```

The parameters can not be recognized by deeptools.

**Comment by baijiexu** [comment]
Hi Fulong,

I update deeptools to v3.5.1. And run the pipeline again. Still not work.

```
[info] Your data won't be shifted as the experiment_type is specified as CUT&RUN...
[info] Peak calling using MACS2... GATA1_D7_30min_chr11.bam
[info] Logs are stored in /home2/s207732/CUT_RUNTools-2.0/bulk-example-test/GATA1_D7_30min_chr11/logs
Fri Jul 22 20:12:39 CDT 2022
[info] Peak calling with BAM file with NO duplications
[info] macs2 narrow peak calling
[info] macs2 broad peak calling
[info] Getting broad peak summits
[info] SEACR stringent peak calling
Calling enriched regions without control file
Proceeding without normalization of control to experimental bedgraph
Using stringent threshold
Creating experimental AUC file: Fri Jul 22 20:12:40 CDT 2022
Calculating optimal AUC threshold: Fri Jul 22 20:12:40 CDT 2022
Using user-provided threshold: Fri Jul 22 20:12:40 CDT 2022
Creating thresholded feature file: Fri Jul 22 20:12:40 CDT 2022
Empirical false discovery rate = 0.01
Merging nearby features and eliminating control-enriched features: Fri Jul 22 20:12:40 CDT 2022
Removing temporary files: Fri Jul 22 20:12:40 CDT 2022
Done: Fri Jul 22 20:12:40 CDT 2022
[info] Generating the normalized signal file with BigWig format...
Fri Jul 22 20:12:40 CDT 2022
cp: cannot stat ‘/home2/s207732/CUT_RUNTools-2.0/bulk-example-test/GATA1_D7_30min_chr11/peakcalling/macs2.narrow/GATA1_D7_30min_chr11.cpm.norm.bw’: No such file or directory
cp: cannot stat ‘/home2/s207732/CUT_RUNTools-2.0/bulk-example-test/GATA1_D7_30min_chr11/peakcalling/macs2.narrow/GATA1_D7_30min_chr11.cpm.norm.bw’: No such file or directory
```
It seems that bam files are generated. macs2 can work well. However, it could not generate bigwig file```` I really don't know what's going on here.

Thank you so much~~~

**Comment by fl-yu** [comment]
can you try deeptools v3.5.0?

**Comment by fl-yu** [comment]
I used this version and I wonder if they make some updates on the normalization parameter in v3.5.1

**Comment by fl-yu** [comment]
Can you try to replace `CPM` with `RPKM` if it still does not work

**Comment by baijiexu** [comment]
I figured it out. Thank you so much, Fulong. I installed the deeptools of v3.5.0. Then it works well.

**Comment by ChristianRohde** [comment]
Hi, I am sorry to bump into this old issue, but I run into the same issue and needed to fix it. Actually I like to find answers this way myself, therefore here is my solution: 

My version of deeptools 3.5.3 which I installed with Python 3.6.15 into the **conda env cutruntools2.1** had the same problem. I confirmed manually on the terminal that indeed it did not work, also editing some parameters did not help.

As a solution I added deeptools to the Python 3.7.12 **conda env meme**. Here deeptools 3.5.4 worked without an problem and further parameter change. 

Best,
Christian
